lms/djangoapps/grades/models.py

```diff
@@ -373,95 +361,50 @@ def update_or_create_grade(cls, **params):
         """
         Wrapper for objects.update_or_create.
         """
-        cls._prepare_params_and_visible_blocks(params)
+        cls._prepare_params(params)
+        VisibleBlocks.cached_get_or_create(params['visible_blocks'])
+        cls._prepare_params_visible_blocks_id(params)
+        cls._prepare_params_override(params)
 
         first_attempted = params.pop('first_attempted')
         user_id = params.pop('user_id')
         usage_key = params.pop('usage_key')
 
-        # apply grade override if one exists before saving model
-        try:
-            override = PersistentSubsectionGradeOverride.objects.get(
-                grade__user_id=user_id,
-                grade__course_id=usage_key.course_key,
-                grade__usage_key=usage_key,
-            )
-            if override.earned_all_override is not None:
-                params['earned_all'] = override.earned_all_override
-            if override.possible_all_override is not None:
-                params['possible_all'] = override.possible_all_override
-            if override.earned_graded_override is not None:
-                params['earned_graded'] = override.earned_graded_override
-            if override.possible_graded_override is not None:
-                params['possible_graded'] = override.possible_graded_override
-        except PersistentSubsectionGradeOverride.DoesNotExist:
-            pass
```

Comment:
**Note to reviewers:** Just a simple refactor to move this to a new reusable helper function: `_prepare_params_override`.